core/core.go

```diff
@@ -48,7 +48,7 @@ func (c *Core) InsertChain(blocks types.Blocks) (int, error) {
 		if !isCoincident && !domWait {
 			err := c.sl.Append(block.Header(), types.EmptyHeader(), common.Hash{}, big.NewInt(0), false, true)
 			if err != nil {
-				if err == consensus.ErrFutureBlock {
+				if err == consensus.ErrFutureBlock || err.Error() == "unknown ancestor" {
```

Comment:
Cannot compare string literals like this. Need to perform string compare, or use a globally shared set of error types... and even then, the latter is error prone.

Reply:
This error is propagated from the sub, so the ethereum native error checking doesn't work. since they use a pointer compare.

Reply:
nvm, I forgot golang supports array deep comparison with the `==` op